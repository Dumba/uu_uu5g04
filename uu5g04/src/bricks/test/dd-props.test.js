/**
 * Copyright (C) 2021 Unicorn a.s.
 *
 * This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public
 * License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License at
 * <https://gnu.org/licenses/> for more details.
 *
 * You may obtain additional information at <https://unicorn.com> or contact Unicorn a.s. at address: V Kapslovne 2767/2,
 * Praha 3, Czech Republic or at the email: info@unicorn.com.
 */

import UU5 from "uu5g04";
import "uu5g04-bricks";

const { mount, shallow, wait } = UU5.Test.Tools;

const CONFIG = {
  mixins: ["UU5.Common.BaseMixin", "UU5.Common.ElementaryMixin", "UU5.Common.ContentMixin"],
  props: {},
  requiredProps: {},
  opt: {
    shallowOpt: {
      disableLifecycleMethods: false,
    },
  },
};

describe(`UU5.Bricks.Dd`, () => {
  UU5.Test.Tools.testProperties(UU5.Bricks.Dd, CONFIG);
});

describe(`UU5.Bricks.Dd is inner bricks.dt`, () => {
  it(`Bricks.Dt is root to: UU5.Bricks.Dd`, () => {
    const wrapper = shallow(
      <UU5.Bricks.Dl id={"uuID01"}>
        <UU5.Bricks.Dt id={"uuID02"} content="Dwarf" />
        <UU5.Bricks.Dd id={"uuID03"} content="The small man with funny hat" />
        <UU5.Bricks.Dt id={"uuID04"} content="Red Dwarf" />
        <UU5.Bricks.Dd id={"uuID05"} content="Small star, or large space ship from british sitcom." />
      </UU5.Bricks.Dl>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
