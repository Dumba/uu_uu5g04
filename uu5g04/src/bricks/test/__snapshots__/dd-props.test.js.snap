// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.Dd checkGrammar=false 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd checkGrammar=true 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd checkHighlight=false 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd checkHighlight=true 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd checkSpaces=false 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd checkSpaces=true 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd className="my-classname" 1`] = `
<dd
  className="uu5-bricks-dd my-classname"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd content="This is Content from ContentMixin" 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
>
  This is Content from ContentMixin
</dd>
`;

exports[`UU5.Bricks.Dd content=12345 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
>
  12345
</dd>
`;

exports[`UU5.Bricks.Dd controlled=false 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd controlled=true 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd default props 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd disabled=false 1`] = `
<dd
  className="uu5-bricks-dd"
  disabled={false}
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd disabled=true 1`] = `
<dd
  className="uu5-bricks-dd"
  disabled={true}
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd dynamic=false 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd dynamic=true 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd hidden=false 1`] = `
<dd
  className="uu5-bricks-dd"
  hidden={false}
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd hidden=true 1`] = `
<dd
  className="uu5-bricks-dd"
  hidden={true}
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd id="myId" 1`] = `
<dd
  className="uu5-bricks-dd"
  id="myId"
/>
`;

exports[`UU5.Bricks.Dd ignoreHTML=false 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd ignoreHTML=true 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd is inner bricks.dt Bricks.Dt is root to: UU5.Bricks.Dd 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID01"
  style={Object {}}
  title={null}
>
  <Dt
    content="Dwarf"
    id="uuID02"
    key="uuID02"
  />
  <Dd
    content="The small man with funny hat"
    id="uuID03"
    key="uuID03"
  />
  <Dt
    content="Red Dwarf"
    id="uuID04"
    key="uuID04"
  />
  <Dd
    content="Small star, or large space ship from british sitcom."
    id="uuID05"
    key="uuID05"
  />
</dl>
`;

exports[`UU5.Bricks.Dd mainAttrs=undefined 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd name="myName" 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd noIndex=false 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd noIndex=true 1`] = `
<dd
  className="uu5-bricks-dd uu5-noindex"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd selected=false 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd selected=true 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd style="padding:8px; borderRadius: 2px" 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
  style={
    Object {
      "borderRadius": "2px",
      "padding": "8px",
    }
  }
/>
`;

exports[`UU5.Bricks.Dd style={"backgroundColor":"red","color":"blue"} 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
  style={
    Object {
      "backgroundColor": "red",
      "color": "blue",
    }
  }
/>
`;

exports[`UU5.Bricks.Dd textCorrector=false 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd textCorrector=true 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;

exports[`UU5.Bricks.Dd tooltip="My Tooltip" 1`] = `
<dd
  className="uu5-bricks-dd"
  id="uuID"
/>
`;
